azext_edge/tests/edge/adr/test_namespace_asset_datasets_int.py

```diff
@@ -645,3 +645,156 @@ def test_namespace_rest_asset_dataset_lifecycle_operations(require_init, tracked
+        f"--instance {instance_name} -g {resource_group} --name {dataset_name_1}"
+    )
+
+    assert_dataset_properties(
+        dataset_show,
+        name=dataset_name_1,
+        data_source=dataset_data_source,
+        asset_type="sse",
+    )
+
+    # 4. UPDATE DATASET
+    updated_destinations = "topic=factory/sse/temperature_v2 qos=Qos0 retain=Never ttl=1800"
+
+    updated_dataset = run(
+        f"az iot ops ns asset sse dataset update --asset {asset_name} "
+        f"--instance {instance_name} -g {resource_group} --name {dataset_name_1} "
+        f"--destination {updated_destinations}"
+    )
+
+    assert_dataset_properties(
+        updated_dataset,
+        name=dataset_name_1,
+        asset_type="sse",
+    )
+
+    # 5. TEST DATASET REPLACE FUNCTIONALITY
+    # Replace dataset with --replace flag
+    replaced_data_source = "/events/temperature/replaced"
+    broker_destinations = "key=sse-data-cache"
+
+    replaced_dataset = run(
+        f"az iot ops ns asset sse dataset add --asset {asset_name} "
+        f"--instance {instance_name} -g {resource_group} --name {dataset_name_1} "
+        f"--data-source {replaced_data_source} --dest {broker_destinations} "
+        f"--replace"
+    )
+
+    assert_dataset_properties(
+        replaced_dataset,
+        name=dataset_name_1,
+        asset_type="sse",
+    )
+
+    # 6. SHOW REPLACED DATASET
+    replaced_dataset_show = run(
+        f"az iot ops ns asset sse dataset show --asset {asset_name} "
+        f"--instance {instance_name} -g {resource_group} --name {dataset_name_1}"
+    )
+
+    assert_dataset_properties(
+        replaced_dataset_show,
+        name=dataset_name_1,
+        data_source=replaced_data_source,
+        asset_type="sse",
+    )
+
+    # 7. CREATE ADDITIONAL DATASET
+    dataset_name_2 = f"dataset2{generate_random_string(6, force_lower=True)}"
+    dataset_2_destinations = "topic=factory/sse/pressure qos=Qos1 retain=Keep ttl=7200"
+
+    dataset_2_result = run(
+        f"az iot ops ns asset sse dataset add --asset {asset_name} "
+        f"--instance {instance_name} -g {resource_group} --name {dataset_name_2} "
+        f"--data-source /events/pressure "
+        f"--destination {dataset_2_destinations}"
+    )
+
+    assert_dataset_properties(
+        dataset_2_result,
+        name=dataset_name_2,
+        data_source="/events/pressure",
+        asset_type="sse",
+    )
+
+    # 8. REMOVE DATASET
+    run(
+        f"az iot ops ns asset sse dataset remove --asset {asset_name} "
+        f"--instance {instance_name} -g {resource_group} --name {dataset_name_1}"
+    )
+
+    # Verify dataset removal
+    datasets_list_after_remove = run(
+        f"az iot ops ns asset sse dataset list --asset {asset_name} "
+        f"--instance {instance_name} -g {resource_group}"
+    )
+
+    remaining_dataset_names = [dataset["name"] for dataset in datasets_list_after_remove]
+    assert dataset_name_1 not in remaining_dataset_names
```

Comment:
do we need to also delete the device? if this runs against a cluster that's not part of a test pipeline, will this leave collateral on the cluster / instance?